atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/SqliteOracleAdapter.java

```diff
@@ -0,0 +1,309 @@
+            insertMetadata.setString(1, tableReference.getQualifiedName());
+            insertMetadata.setInt(2, 1);
+            insertMetadata.executeUpdate();
+
+            PreparedStatement insertTableMapping = connection.prepareStatement("INSERT INTO "
+                    + AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE + " (table_name, short_table_name) VALUES (?, ?)");
+            insertTableMapping.setString(1, reversibleTableName);
+            insertTableMapping.setString(2, physicalTableName);
+            insertTableMapping.executeUpdate();
+            return TableDetails.builder()
+                    .tableReference(tableReference)
+                    .physicalTableName(physicalTableName)
+                    .reversibleTableName(reversibleTableName)
+                    .build();
+        });
+    }
+
+    public Set<String> listAllTables() {
+        refreshTableList();
+        return runWithConnection(connection -> {
+            PreparedStatement statement = connection.prepareStatement("SELECT name FROM sqlite_schema WHERE"
+                    + " type ='table' AND name NOT LIKE 'sqlite_%' AND name != 'all_tables' AND"
+                    + " name != ? AND name != ?");
+            statement.setString(1, METADATA_TABLE.getQualifiedName());
+            statement.setString(2, AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE);
+            ResultSet resultSet = statement.executeQuery();
+
+            ImmutableSet.Builder<String> tableNames = ImmutableSet.builder();
+            while (resultSet.next()) {
+                tableNames.add(resultSet.getString("name"));
+            }
+            return tableNames.build();
+        });
+    }
+
+    // all_tables is not a native sqlite concept, unlike Oracle. Thus, if we perform deletes, we don't cascade and
+    // delete from all_tables. This method does not insert tables created outside the create_table method
+    private void refreshTableList() {
+        runWithConnection(connection -> {
+            connection
+                    .createStatement()
+                    .executeUpdate("DELETE FROM all_tables WHERE lower(table_name) NOT IN (SELECT lower(name) as"
+                            + " table_name FROM sqlite_schema WHERE type = 'table')");
+            return null;
+        });
+    }
+
+    public Set<TableReference> listAllTablesWithMetadata() {
+        return runWithConnection(connection -> {
+            PreparedStatement statement = connection.prepareStatement("SELECT table_name FROM " + METADATA_TABLE);
+            ResultSet resultSet = statement.executeQuery();
+
+            ImmutableSet.Builder<TableReference> tableNames = ImmutableSet.builder();
+            while (resultSet.next()) {
+                tableNames.add(TableReference.createFromFullyQualifiedName(resultSet.getString("table_name")));
+            }
+            return tableNames.build();
+        });
+    }
+
+    public Set<String> listAllTablesWithMapping() {
```

Comment:
Do we not want to return `Set<TableReference>` here?

Reply:
Not particularly, this is used for comparing that the tables in the mapping table are the same as the tables we expect to be in the mapping table. I _can_ return table reference, but that requires unmapping the table names for little benefit.